shipyard_blueprints/hubspot/shipyard_hubspot/hubspot.py

```diff
@@ -1,34 +1,74 @@
+import os
 import json
 
 from logging import DEBUG
+from typing import Dict, List, Union, Optional
+
 from requests import request
 from shipyard_templates import Crm, ExitCodeException
+from shipyard_hubspot.hubspot_utils import (
+    handle_request_errors,
+    validate_export_type,
+    validate_import_operations,
+    validate_export_language,
+    validate_date_format,
+    validate_export_file_format,
+    handle_import_file,
+)
 
 
 class HubspotClient(Crm):
-    def __init__(self, access_token: str, verbose: bool = False) -> None:
-        self.access_token = access_token
+    """
+    A client for interacting with the Hubspot API.
 
-        super().__init__(access_token)
+    Attributes:
+        access_token (str): The access token used for authentication.
+    """
+
+    def __init__(self, access_token: str, verbose: bool = False):
+        """
+        Initialize the HubspotClient.
 
+        :param access_token: The access token used for authentication.
+        :param verbose: If True, sets the logger to debug mode.
+        """
+        self.access_token = access_token
+        super().__init__(access_token)
         self.logger.info("HubspotClient initialized")
         if verbose:
             self.logger.info("Verbose mode enabled")
             self.logger.setLevel(DEBUG)
+            for handler in self.logger.handlers:
+                handler.setLevel(DEBUG)
 
-    def _requests(self, endpoint: str, method: str = "GET", payload: dict = None):
+    def _requests(
+        self,
+        endpoint: str,
+        method: str = "GET",
+        payload: Optional[Dict] = None,
+        headers: Optional[Dict] = None,
```

Comment:
Any reason why we don't set the headers as an attribute for the class?

Reply:
I believe keeping it in the helper function makes it slightly more flexible. Not all of the endpoints accept the same headerdata- For example the import endpoint would fail if  tried to pass in the default header. it is a bad example considering that i opted to bypass the helper function but is an example of the headers needing some flexibility. In hindsight i should set the default value in the function instead of handling it within the script.

Not sure what the benefit would be to change it into the class attribute.